spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -189,12 +189,13 @@ private static void mergeIcebergHadoopConfs(
         .forEach(key -> baseConf.set(key.replaceFirst("hadoop.", ""), options.get(key)));
   }
 
-  private void validateWriteSchema(Schema tableSchema, Schema dsSchema, Boolean checkNullability) {
+  private void validateWriteSchema(
+          Schema tableSchema, Schema dsSchema, Boolean checkNullability, Boolean checkOrdering) {
```

Comment:
It looks like this is getting too complicated. We now have 2 boolean options, one of which is used to select the compatibility checking method (which have similar names) and the other is passed as an arg (which isn't readable). I think it is a good idea to convert compatibility checking to a builder-like pattern:

```java
CheckCompatibility
    .writeSchema(dsSchema)
    .readSchema(tableSchema)
    .checkOrdering(true)
    .checkNullability(false)
    .throwOnValidationError();
```